src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -921,45 +921,55 @@ public static bool IsValidParams(Symbol member)
             return final.IsParams && ((ParameterSymbol)final.OriginalDefinition).Type.IsSZArray();
         }
 
-        private static bool IsOverride(Symbol overridden, Symbol overrider)
+        /// <summary>
+        /// Does <paramref name="moreDerivedOverride"/> override <paramref name="member"/> or the
+        /// thing that it originally overrides, but in a more derived class?
+        /// </summary>
+        private static bool IsMoreDerivedOverride(Symbol member, Symbol moreDerivedOverride)
         {
-            if (TypeSymbol.Equals(overridden.ContainingType, overrider.ContainingType, TypeCompareKind.ConsiderEverything2) ||
-                !MemberSignatureComparer.SloppyOverrideComparer.Equals(overridden, overrider))
+            if (!moreDerivedOverride.IsOverride ||
+                !IsBaseClass(derivedType: moreDerivedOverride.ContainingType, baseType: member.ContainingType) ||
+                !MemberSignatureComparer.SloppyOverrideComparer.Equals(member, moreDerivedOverride))
             {
                 // Easy out.
                 return false;
             }
 
-            // Does overrider override overridden?
-            var current = overrider;
-            while (true)
-            {
-                if (!current.IsOverride)
-                {
-                    return false;
-                }
-                current = current.GetOverriddenMember();
+            // Rather than following the member.GetOverriddenMember() chain, we check to see if both
+            // methods ultimately override the same original method.  This addresses issues in binary compat
+            // scenarios where the override chain may skip some steps.
+            // See https://github.com/dotnet/roslyn/issues/45798 for an example.
+            return moreDerivedOverride.GetLeastOverriddenMember(accessingTypeOpt: null).OriginalDefinition ==
+                   member.GetLeastOverriddenMember(accessingTypeOpt: null).OriginalDefinition;
+        }
 
-                // We could be in error recovery.
-                if ((object)current == null)
-                {
-                    return false;
-                }
+        private static bool IsBaseClass(TypeSymbol derivedType, TypeSymbol baseType)
```

Comment:
>IsBaseClass [](start = 28, length = 11)

It looks like this is a dupe of ```TypeSymbol.IsDerivedFrom``` method.

Reply:
It is similar. It is closer to ConversionBase.IsBaseClass.

---
In reply to: [461762763](https://github.com/dotnet/roslyn/pull/46367#discussion_r461762763) [](ancestors = 461762763)

Reply:
> It is similar. It is closer to ConversionBase.IsBaseClass.

It is quite possible I am missing something, but based on a quick glance at the code in both methods, I see no meaningful difference. Please, point me to a unit-test that would demonstrate an observable difference between using this method and using ```TypeSymbol.IsDerivedFrom```  method in ```IsMoreDerivedOverride```, or add a test like that. Otherwise, I believe there is no benefit in cloning existing helpers and duplicating code in the product, and we should simply switch to using ```TypeSymbol.IsDerivedFrom```.

---
In reply to: [461831483](https://github.com/dotnet/roslyn/pull/46367#discussion_r461831483) [](ancestors = 461831483,461762763)

Reply:
This code is specialized to avoid gathering use site diagnostics and to perform an exact type check. If you don't mind the extra work of computing and discarding use-site diagnostics I will reuse that.

---
In reply to: [461944824](https://github.com/dotnet/roslyn/pull/46367#discussion_r461944824) [](ancestors = 461944824,461831483,461762763)